packages/api/src/utils/auth.ts

```diff
@@ -1,16 +1,51 @@
 import * as bcrypt from 'bcryptjs'
-import * as jwt from 'jsonwebtoken'
-import { env } from '../env'
+import { v4 as uuidv4 } from 'uuid'
 import { Claims } from '../resolvers/types'
+import { getRepository } from '../entity/utils'
+import { ApiKey } from '../entity/api_key'
+import crypto from 'crypto'
 
-export const hashPassword = (password: string) => {
-  return bcrypt.hashSync(password, 10)
+export const hashPassword = async (password: string, salt = 10) => {
+  return bcrypt.hash(password, salt)
 }
 
-export const comparePassword = (password: string, hash: string) => {
-  return bcrypt.compareSync(password, hash)
+export const comparePassword = async (password: string, hash: string) => {
+  return bcrypt.compare(password, hash)
 }
 
-export const generateApiKey = (claims: Claims): string => {
-  return jwt.sign(claims, env.server.jwtSecret)
+export const generateApiKey = (): string => {
+  // TODO: generate random string key
+  return uuidv4()
+}
+
+export const hashApiKey = (apiKey: string) => {
+  return crypto.createHash('sha256').update(apiKey).digest('hex')
+}
+
+export const claimsFromApiKey = async (key: string): Promise<Claims> => {
+  const hashedKey = hashApiKey(key)
```

Comment:
Do we need to do this hash ourselves, can we do something like this? https://www.meetspaceapp.com/2016/04/12/passwords-postgresql-pgcrypto.html

Reply:
Yeah I think we can. 
Personally I prefer doing hashing/encryption (also the default value adding and unique checking) at the application level in case we need to migrate to another database like mysql. 
LOL. I guess I have PTSD

Reply:
hahah we would never migrate to another data....oh wait I guess we did already.